api/src/main/java/io/opentelemetry/common/Acceptor.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.common;
+
+/**
+ * A function to be executed with a single argument, having a side effect.
+ *
+ * <p>Adapted from Java 8's {@code java.util.function.Consumer}.
+ *
+ * @param <T> Type of the input argument.
+ */
+public interface Acceptor<T> {
+
+  /**
+   * Performs this operation on the given argument.
+   *
+   * @param arg the input argument
+   */
+  void exec(T arg);
```

Comment:
On an interface called `Acceptor`, I'd expect this method to be called `accept`.

Reply:
thinking out loud here, but what if we had this return `T`, rather than `void`, making it feel more like a way to extend the builder fluent nature?